core/lib/dal/src/protocol_versions_dal.rs

```diff
@@ -29,32 +29,63 @@ pub struct ProtocolVersionsDal<'a, 'c> {
 impl ProtocolVersionsDal<'_, '_> {
     pub async fn save_protocol_version(
         &mut self,
-        id: ProtocolVersionId,
+        version: ProtocolSemanticVersion,
         timestamp: u64,
         l1_verifier_config: L1VerifierConfig,
         base_system_contracts_hashes: BaseSystemContractsHashes,
         tx_hash: Option<H256>,
     ) -> DalResult<()> {
+        let mut db_transaction = self.storage.start_transaction().await?;
+
         sqlx::query!(
             r#"
             INSERT INTO
                 protocol_versions (
                     id,
                     timestamp,
-                    recursion_scheduler_level_vk_hash,
```

Comment:
just in case, will the migration cause a small downtime? I.e. the old SQL does not work with the previous DB state and the new one does not work with the old one

Reply:
If I haven't missed anything, then the only query that breaks is the one that inserts protocol versions. Given that we won't have upgrades in between applying migrations and updating server it won't cause downtime. Please raise if you think some other query may break and I missed it